### PR TITLE
Improve docs and type hints

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
# What does this PR do?

Part of #5249.

Improves type hints, docs and minor refactors in the following files:
- [examples/text_to_image/train_text_to_image_flax.py](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_flax.py) (result of `make style`)
- [examples/text_to_image/train_text_to_image_lora_sdxl.py](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_lora_sdxl.py) (result of `make style`)
- [src/diffusers/models/autoencoder_asym_kl.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/autoencoder_asym_kl.py)
- [src/diffusers/models/autoencoder_kl.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/autoencoder_kl.py)
- [src/diffusers/models/autoencoder_tiny.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/autoencoder_tiny.py)
- [src/diffusers/models/consistency_decoder_vae.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/consistency_decoder_vae.py)
- [src/diffusers/models/controlnet.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/controlnet.py)
- [src/diffusers/models/controlnet_flax.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/controlnet_flax.py)
- [src/diffusers/models/modeling_utils.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/modeling_utils.py)
- [src/diffusers/models/normalization.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/normalization.py)
- [src/diffusers/models/resnet.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/resnet.py)
- [src/diffusers/models/unet_2d_condition_flax.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unet_2d_condition_flax.py)
- [src/diffusers/models/unet_3d_blocks.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unet_3d_blocks.py)
- [src/diffusers/models/unet_3d_condition.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unet_3d_condition.py)
- [src/diffusers/models/unet_motion_model.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unet_motion_model.py)
- [src/diffusers/models/vq_model.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/vq_model.py)
- [src/diffusers/image_processor.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/image_processor.py)
- [src/diffusers/optimization.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/optimization.py)
- [src/diffusers/utils/logging.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/utils/logging.py)
- [src/diffusers/utils/outputs.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/utils/outputs.py)
- [src/diffusers/utils/torch_utils.py](https://github.com/huggingface/diffusers/blob/main/src/diffusers/utils/torch_utils.py)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@DN6 @yiyixuxu

### CODE REVIEW

File: src/diffusers/models/autoencoder_asym_kl.py [hunk 1]
```diff
@@ -125,7 +127,7 @@ def _decode(
         image: Optional[torch.FloatTensor] = None,
         mask: Optional[torch.FloatTensor] = None,
         return_dict: bool = True,
-    ) -> Union[DecoderOutput, torch.FloatTensor]:
+    ) -> Union[DecoderOutput, Tuple[torch.FloatTensor]]:
```
Comment (DN6) [inline src/diffusers/models/autoencoder_asym_kl.py:130]: I think the `decode` method has the same type hints too

File: src/diffusers/models/controlnet.py [hunk 2]
```diff
@@ -774,7 +777,7 @@ def forward(
         sample = self.conv_in(sample)
 
         controlnet_cond = self.controlnet_cond_embedding(controlnet_cond)
-        sample = sample + controlnet_cond
+        sample += controlnet_cond
```
Comment (DN6) [inline src/diffusers/models/controlnet.py:780]: Not a type hint/docs change. Maybe we can open a separate PR for clean up ops like this?

**Review by DN6 — COMMENTED** [review]
Very nicely done. Just had a few small suggestions.

File: src/diffusers/models/controlnet.py (hunk 2 above)
Comment (a-r-r-o-w) [reply src/diffusers/models/controlnet.py:780]: Sure! Will undo such changes

**Comment by a-r-r-o-w** [comment]
@DN6 @yiyixuxu I believe this is ready for another review. Will add more changes in upcoming PRs  :rocket:

**Review by DN6 — APPROVED** [review]

**Pull request merged**